frontend/app/user/people-list.component.js

```diff
@@ -32,6 +32,20 @@ export default class PeopleList extends React.Component {
     });
   };
 
+  deleteCase = (person, kase, e) => {
```

Comment:
This function could be simplified to the following:

```js
deleteCase = (person, kase, e) => {
  e.preventDefault()
  this.setState(prevState => ({
    cases: prevState.cases.filter(c => c !== kase)
  }, () => {
    database.deleteCase(kase.id)
  })
}
```

You don't need lodash's remove function, you can just use [Array.prototype.filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter). Additionally, in React class components you should use `prevState` with an updater function instead of `this.state` [as described here](https://reactjs.org/docs/react-component.html#setstate) when updating state based on it's previous value.

Reply:
> cases: prevState.cases.filter(c => c !== kase) makes it look like the state is being changed directly, but it isn't, because it's in setState method.

No. The reason has nothing to do with it being inside of a setState method. It has everything to do with Array.prototype.filter creating a **new array** without modifying the existing one. You can use filter safely because it doesn't mutate the array